substrate/frame/support/procedural/examples/proc_main/main.rs

```diff
@@ -0,0 +1,739 @@
+#[cfg(test)]
+use sp_io::{MultiRemovalResults, TestExternalities};
+#[cfg(test)]
+use sp_metadata_ir::{
+	PalletStorageMetadataIR, StorageEntryMetadataIR, StorageEntryModifierIR, StorageEntryTypeIR,
+	StorageHasherIR,
+};
+#[cfg(test)]
+use sp_runtime::BuildStorage;
+use sp_runtime::{generic, traits::BlakeTwo256};
+
+pub use self::frame_system::{pallet_prelude::*, Config, Pallet};
+
+mod inject_runtime_type;
+mod runtime;
+mod tasks;
+
+#[import_section(tasks::tasks_example)]
+#[pallet]
+pub mod frame_system {
+	#[allow(unused)]
+	use super::{frame_system, frame_system::pallet_prelude::*};
+	pub use crate::dispatch::RawOrigin;
+	use crate::{pallet_prelude::*, traits::tasks::Task as TaskTrait};
+
+	pub mod config_preludes {
+		use super::{inject_runtime_type, DefaultConfig};
+		pub struct TestDefaultConfig;
+
+		#[crate::register_default_impl(TestDefaultConfig)]
+		impl DefaultConfig for TestDefaultConfig {
+			type AccountId = u64;
+			type BaseCallFilter = frame_support::traits::Everything;
+			#[inject_runtime_type]
+			type RuntimeOrigin = ();
+			#[inject_runtime_type]
+			type RuntimeCall = ();
+			#[inject_runtime_type]
+			type PalletInfo = ();
+			#[inject_runtime_type]
+			type RuntimeTask = ();
+			type DbWeight = ();
+		}
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config(with_default)]
+	#[pallet::disable_frame_system_supertrait_check]
+	pub trait Config: 'static {
+		#[pallet::no_default]
+		type Block: Parameter + sp_runtime::traits::Block;
+		type AccountId;
+		#[pallet::no_default_bounds]
+		type BaseCallFilter: crate::traits::Contains<Self::RuntimeCall>;
+		#[pallet::no_default_bounds]
+		type RuntimeOrigin;
+		#[pallet::no_default_bounds]
+		type RuntimeCall;
+		#[pallet::no_default_bounds]
+		type RuntimeTask: crate::traits::tasks::Task;
+		#[pallet::no_default_bounds]
+		type PalletInfo: crate::traits::PalletInfo;
+		type DbWeight: Get<crate::weights::RuntimeDbWeight>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Required by construct_runtime
+		CallFiltered,
+		/// Used in tasks example.
+		NotFound,
+		/// The specified [`Task`] is not valid.
+		InvalidTask,
+		/// The specified [`Task`] failed during execution.
+		FailedTask,
+	}
+
+	#[pallet::origin]
+	pub type Origin<T> = RawOrigin<<T as Config>::AccountId>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight(task.weight())]
+		pub fn do_task(_origin: OriginFor<T>, task: T::RuntimeTask) -> DispatchResultWithPostInfo {
+			if !task.is_valid() {
+				return Err(Error::<T>::InvalidTask.into())
+			}
+
+			if let Err(_err) = task.run() {
+				return Err(Error::<T>::FailedTask.into())
+			}
+
+			Ok(().into())
+		}
+	}
+
+	#[pallet::storage]
+	pub type Data<T> = StorageMap<_, Twox64Concat, u32, u64, ValueQuery>;
+
+	#[pallet::storage]
+	pub type OptionLinkedMap<T> = StorageMap<_, Blake2_128Concat, u32, u32, OptionQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn generic_data)]
```

Comment:
Shouldn't we stop using getters since we're deprecating them?

Reply:
Yeah, this file is just moved from the support crate but let me remove this nonetheless.